src/Components/Location.jsx

```diff
@@ -0,0 +1,49 @@
+import React, { useEffect } from 'react'
+import { useSelector, useDispatch } from 'react-redux'
+import axios from 'axios'
+
+const Location = () => {
+  const { userCountry } = useSelector((state) => state)
+  const dispatch = useDispatch()
+
+  const fetchUserCoordinates = async () => {
+    return new Promise((resolve, reject) => {
+      navigator.geolocation.getCurrentPosition(resolve, reject, {
+        enableHighAccuracy: true,
+      })
+    })
+  }
+
+  useEffect(() => {
+    fetchGeolocation()
+  }, [])
+
+  const fetchGeolocation = async () => {
+    const position = await fetchUserCoordinates()
+
+    if (position.error) {
+      dispatch({
+        type: 'SET_USER_COUNTRY',
+        payload: 'location cannot be detected',
+      })
```

Comment:
/Users/gavinsimpson/dgms-news-end-user-client/cypress/integration/userCanAccessArticlesBasedOnLocation.feauture.js
/79

  it('is expected to show that the users location is not found', () => {
    cy.get('[data-cy=user-location]').should(
      'contain',
      'location cannot be detected',

Does this test? (have reworded since original PR)

Reply:
Yes! Good stuff.